server/service/src/invoice_line/stock_out_line/validate.rs

```diff
@@ -43,42 +43,36 @@ pub fn check_unique_stock_line(
     }
 }
 
-pub fn check_item_matches_batch(batch: &StockLineRow, item: &ItemRow) -> bool {
-    if batch.item_id != item.id {
+pub fn check_item_matches_batch(batch: &StockLine, item: &ItemRow) -> bool {
+    if batch.item_row.id != item.id {
         return false;
     }
     return true;
 }
 
-pub fn check_batch_on_hold(batch: &StockLineRow) -> bool {
-    if batch.on_hold {
+pub fn check_batch_on_hold(batch: &StockLine) -> bool {
+    if batch.stock_line_row.on_hold {
         return false;
     }
     return true;
 }
 
 pub enum LocationIsOnHoldError {
     LocationIsOnHold,
-    LocationNotFound,
 }
 
 pub fn check_location_on_hold(
-    batch: &StockLineRow,
+    batch: &StockLine,
     connection: &StorageConnection,
 ) -> Result<(), LocationIsOnHoldError> {
     use LocationIsOnHoldError::*;
 
-    match &batch.location_id {
-        Some(location_id) => {
-            let location = LocationRowRepository::new(connection)
-                .find_one_by_id(&location_id)
-                .map_err(|_| LocationNotFound)?;
-
-            match location {
-                Some(location) if location.on_hold => Err(LocationIsOnHold),
-                Some(_) => Ok(()),
-                None => Err(LocationNotFound),
+    match &batch.location_row {
+        Some(location) => {
+            if location.on_hold {
+                return Err(LocationIsOnHold);
```

Comment:
Since the StockLine already has the location loaded I was able to get rid of the LocationNotFound error :)

Reply:
This method could be refactored to just return a `bool` but I leave it for now...